pkg/local/sync.go

```diff
@@ -105,6 +93,9 @@ func (s *SyncManager) Sync(rootPath string, remote *uri.URI, changeSet <-chan *C
 }
 
 func (s *SyncManager) apply(ctx context.Context, rootPath string, remote *uri.URI, change *Change) (err error) {
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
```

Comment:
capture the err once some implementations work "harder" to get the err value
```suggestion
	if err := ctx.Err(); err != nil {
		return err
	}
```